controlplane/kubeadm/internal/workload_cluster_etcd.go

```diff
@@ -116,20 +136,38 @@ func (w *Workload) EtcdIsHealthy(ctx context.Context) (HealthCheckResult, error)
 		} else {
 			unknownMembers := memberIDSet.Difference(knownMemberIDSet)
 			if unknownMembers.Len() > 0 {
+				conditions.MarkFalse(controlPlane.KCP, controlplanev1.EtcdClusterHealthy, controlplanev1.EtcdClusterUnhealthyReason, ConditionReason(controlplanev1.EtcdClusterUnhealthyReason).GetSeverity(), "etcd members do not have the same member-list view")
 				response[name] = errors.Errorf("etcd member reports members IDs %v, but all previously seen etcd members reported member IDs %v", memberIDSet.UnsortedList(), knownMemberIDSet.UnsortedList())
 			}
 			continue
 		}
 	}
 
+	// Check etcd cluster alarms
+	etcdClient, err := w.etcdClientGenerator.forNodes(ctx, controlPlaneNodes.Items)
+	if err == nil {
+		defer etcdClient.Close()
+		alarmList, err := etcdClient.Alarms(ctx)
+		if len(alarmList) > 0 || err != nil {
+			conditions.MarkFalse(controlPlane.KCP, controlplanev1.EtcdClusterHealthy, controlplanev1.EtcdClusterUnhealthyReason, ConditionReason(controlplanev1.EtcdClusterUnhealthyReason).GetSeverity(), "etcd cluster has alarms.")
+			return response, errors.Errorf("etcd cluster has %d alarms", len(alarmList))
+		}
+	}
+
 	// TODO: ensure that each pod is owned by a node that we're managing. That would ensure there are no out-of-band etcd members
 
 	// Check that there is exactly one etcd member for every healthy pod.
 	// This allows us to handle the expected case where there is a failing pod but it's been removed from the member list.
 	if expectedMembers != len(knownMemberIDSet) {
+		conditions.MarkFalse(controlPlane.KCP, controlplanev1.EtcdClusterHealthy, controlplanev1.EtcdClusterUnhealthyReason, ConditionReason(controlplanev1.EtcdClusterUnhealthyReason).GetSeverity(), "etcd pods does not match with etcd members.")
 		return response, errors.Errorf("there are %d healthy etcd pods, but %d etcd members", expectedMembers, len(knownMemberIDSet))
 	}
 
+	// TODO: During provisioning, this condition may be set false for a short time until all pods are provisioned, can add additional checks here to prevent this.
```

Comment:
I saw this condition flake in my test, and this could be really worrying for users.
Wondering about about a possible strategy to avoid this...

Reply:
Although it won't solve it 100%, we can check if there is any Node with `NodeHealthy` condition in `Provisioning` state. 
Another idea is to check the health of other control-plane pods on that Node to get an idea about if it is still in provisioning.

Reply:
Added etcd individual member health check using the same logic with etcdctl. Tries to get a key-value, if not failing, then the member is healthy. 
https://github.com/etcd-io/etcd/blob/ff455e3567d6465717d3ced6dc889d4eec9ed890/etcdctl/ctlv3/command/ep_command.go#L121